pkg/cmd/admin/policy/modify_roles.go

```diff
@@ -218,6 +227,37 @@ func NewCmdRemoveClusterRoleFromUser(name, fullName string, f *clientcmd.Factory
 	return cmd
 }
 
+func (o *RoleModificationOptions) CompleteUserWithSA(f *clientcmd.Factory, args []string, saLong, saShort util.StringList) error {
+	if (len(args) < 2) && (len(saLong) == 0) && (len(saShort) == 0) {
+		return errors.New("You must specify at least two arguments: <role> <user> [user]...")
+	}
+
+	o.RoleName = args[0]
+	if len(args) > 1 {
+		o.Users = append(o.Users, args[1:]...)
+	}
+
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	roleBindingNamespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.RoleBindingAccessor = NewLocalRoleBindingAccessor(roleBindingNamespace, osClient)
+
+	for _, sa := range saLong {
+		o.Users = append(o.Users, fmt.Sprintf("system:serviceaccount:%s:%s", roleBindingNamespace, sa))
```

Comment:
Extract this to a function in authorization/helpers.go so that this isn't in code here.